File: ToolKit/src/main/java/com/uday/android/toolkit/runnable/KernelAction.java
package com.uday.android.toolkit.runnable;


import com.uday.android.toolkit.*;
import com.uday.android.toolkit.fragments.*;
import java.io.*;

public class KernelAction implements Runnable
{
	private KernelFragment fragment;
	public KernelAction(KernelFragment fragment){
		this.fragment=fragment;
	}
	
	
	@Override
	public void run(){
		switch(fragment.mOption){
			case fragment.SELECTED_BACKUP:
				File backup=new File(fragment.DIR.getAbsolutePath()+"/backups/Kernel/"+fragment.kernVersion);
				backup.mkdirs();
				MainActivity.rootSession.addCommand(fragment.context.getFilesDir()+"/common/kern_backup.sh "+MainActivity.TOOL+" "+fragment.BLOCK+" '"+backup.getAbsolutePath()+"'",fragment.SELECTED_BACKUP,fragment.commandLineListener);
				
				break;
			case fragment.SELECTED_RESTORE:
				MainActivity.rootSession .addCommand(fragment.context.getFilesDir()+"/common/kern_install.sh "+MainActivity.TOOL+" "+fragment.BLOCK+" '"+fragment.selected.getAbsolutePath()+"/kernel' false",fragment.SELECTED_RESTORE,fragment.commandLineListener);
				
				break;
			case fragment.SELECTED_INSTALL:
				MainActivity.rootSession.addCommand(fragment.context.getFilesDir()+"/common/kern_install.sh "+MainActivity.TOOL+" "+fragment.BLOCK+" "+'"'+fragment.file.getAbsolutePath()+'"'+" "+fragment.isImg,fragment.SELECTED_INSTALL,fragment.commandLineListener);
				
				break;
		}
	}
	
}

File: app/src/main/java/com/uday/android/toolkit/runnable/KernelAction.kt
package com.uday.android.toolkit.runnable


import android.annotation.SuppressLint
import com.uday.android.toolkit.MainActivity
import com.uday.android.toolkit.fragments.KernelFragment
import java.io.File

class KernelAction(private val fragment: KernelFragment) : Runnable {


    @SuppressLint("NewApi")
    override fun run() {
        when (fragment.mOption) {
            KernelFragment.SELECTED_BACKUP -> {
                val backup =
                    File(fragment.dir!!.absolutePath + "/backups/Kernel/" + fragment.kernVersion)
                backup.mkdirs()
                MainActivity.rootSession!!.addCommand(
                    fragment.context.filesDir.toString() + "/common/kern_backup.sh " + MainActivity.TOOL + " " + fragment.block + " '" + backup.absolutePath + "'",
                    KernelFragment.SELECTED_BACKUP,
                    fragment.commandLineListener
                )
            }
            KernelFragment.SELECTED_RESTORE -> MainActivity.rootSession!!.addCommand(
                fragment.context.filesDir.toString() + "/common/kern_install.sh " + MainActivity.TOOL + " " + fragment.block + " '" + fragment.selected!!.absolutePath + "/kernel' false",
                KernelFragment.SELECTED_RESTORE,
                fragment.commandLineListener
            )
            KernelFragment.SELECTED_INSTALL -> MainActivity.rootSession!!.addCommand(
                fragment.context.filesDir.toString() + "/common/kern_install.sh " + MainActivity.TOOL + " " + fragment.block + " " + '"'.toString() + fragment.file!!.absolutePath + '"'.toString() + " " + fragment.isImg,
                KernelFragment.SELECTED_INSTALL,
                fragment.commandLineListener
            )
        }
    }

}
